tests/sles4sap/mr_test.pm

```diff
@@ -0,0 +1,235 @@
+            @overrides = ("${note}-1", "${note}-2");
+        }
+        if ($note eq "1680803") {
+            # Ignore the test for the scheduler if we can't set "noop" on /dev/sr0
+            assert_script_run(
+                "if [ -f /sys/block/sr0/queue/scheduler ] ; then "
+                  . "grep -q noop /sys/block/sr0/queue/scheduler || "
+                  . "sed -i '/:scripts\\/nr_requests/s/^/#/' Pattern/$SLE/testpattern_note_${note}_a_override ; fi"
+            );
+        }
+        foreach my $override (@overrides) {
+            my @commands = (
+                "mr_test verify Pattern/${SLE}/testpattern_baseline_Cust",
+                "mr_test dump Pattern/$SLE/testpattern_note_${override}_b > baseline_testpattern_note_${override}_b",
+                "cp Pattern/$SLE/override/$override /etc/saptune/override/$note",
+                "saptune note apply $note",
+                "mr_test verify Pattern/$SLE/testpattern_note_${override}_a_override",
+                "mr_test verify baseline_testpattern_note_${override}_b",
+            );
+            foreach my $cmd (@commands) {
+                assert_script_run($cmd);
+            }
+            tune_baseline("baseline_testpattern_note_${override}_b");
+            $self->reboot;
+            @commands = (
+                "mr_test verify Pattern/$SLE/testpattern_note_${override}_a_override",
+                "mr_test verify baseline_testpattern_note_${override}_b",
+                "saptune note revert $note",
+                "rm -f /etc/saptune/override/$note",
+            );
+            foreach my $cmd (@commands) {
+                assert_script_run($cmd);
+            }
+            $self->reboot;
+        }
+    }
+}
+
+sub test_solutions {
+    my ($self) = @_;
+
+    my $SLE = is_sle(">=15") ? "SLE15" : "SLE12";
+
+    foreach my $solution (get_solutions()) {
+        my @commands = (
+            "mr_test verify Pattern/${SLE}/testpattern_baseline_Cust",
+            "mr_test dump Pattern/${SLE}/testpattern_solution_${solution}_b > baseline_testpattern_solution_${solution}_b",
+            "saptune solution apply $solution",
+            "mr_test verify Pattern/${SLE}/testpattern_solution_${solution}_a",
+            "mr_test verify baseline_testpattern_solution_${solution}_b",
+        );
+        foreach my $cmd (@commands) {
+            assert_script_run($cmd);
+        }
+        tune_baseline("baseline_testpattern_solution_${solution}_b");
+        $self->reboot;
+        @commands = (
+            "mr_test verify Pattern/${SLE}/testpattern_solution_${solution}_a",
+            "mr_test verify baseline_testpattern_solution_${solution}_b",
+            "saptune solution revert $solution",
+        );
+        foreach my $cmd (@commands) {
+            assert_script_run($cmd);
+        }
+        $self->reboot;
+    }
+}
+
+sub run {
+    my ($self) = @_;
+
+    $self->setup;
+
+    my $test = get_required_var("MR_TEST");
+    if ($test eq "sapconf") {
```

Comment:
1. I wouldn't mix the `sapconf` test with any other test because any configuration left over by sapconf would mess with saptune, giving us false positives.

2. Running any combination of `solutions`, `notes` & `overrides` takes much too long time and the final consolidation of the screenshots takes a long time too because there are lots of reboots.  Also, I think it's better to have them running separate in case a bug appears only on overrides, for example.

Changing the code would be trivial anyway but with no real gain.